_posts/2020-12-01-configuring-and-suppressing-warnings.md

```diff
@@ -0,0 +1,192 @@
+---
+layout: blog-detail
+post-type: blog
+by: Lukas Rytz
+title: "Configuring and suppressing warnings in Scala"
+isHighlight: true
+---
+
+Scala 2.13.2 and 2.12.13 introduced the `-Wconf` compiler flag to globally configure reporting of warnings, and the `@nowarn` annotation to locally suppress them. Having more control over compiler warnings makes them a lot more valuable:
```

Comment:
I think it would be a good idea to more explicitly call attention to the 2.12 backport, since that's fresh news. "This addition to Scala 2.13 proved very popular, so..."

Reply:
But also, were any corners cut, was anything left out in the 2.12 backport?

Maybe mention that scala-collection-compat includes a no-op `@nowarn` to allow crossbuilding to 2.11?

hey @dwijnand do we need to publish a new version of scala-collection-compat that removes that no-op `@nowarn` from the 2.12 artifact, actually?

Reply:
Yes, best to, seeing as it's properly implemented as a `ClassfileAnnotation` in 2.12.x instead of a `StaticAnnotation` in scala-collection-compat.  Which I know isn't right because I did that too first time round until one of Lukas' tests wasn't working with that.

* https://github.com/scala/scala/blob/2.12.x/src/library/scala/annotation/nowarn.scala
* https://github.com/scala/scala-collection-compat/blob/master/compat/src/main/scala-2.11_2.12/scala/annotation/nowarn.scala

Reply:
https://github.com/scala/scala-collection-compat/issues/397